web/modules/custom/bnf/bnf_client/src/BnfScheduler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\bnf_client;
+
+use Drupal\bnf\BnfStateEnum;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Queue\QueueFactory;
+
+/**
+ * Handles queueing of updates to subscriptions and content.
+ */
+class BnfScheduler {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    protected EntityTypeManagerInterface $entityTypeManager,
+    protected QueueFactory $queueFactory,
+  ) {}
+
+  /**
+   * Queue an update check on all subscriptions.
+   */
+  public function queueAllSubscriptionsUpdate(): void {
+    /** @var \Drupal\bnf_client\Entity\Subscription[] $subscriptions */
+    $subscriptions = $this->entityTypeManager->getStorage('bnf_subscription')->loadMultiple();
```

Comment:
I guess we're not planning on sites having hundreds of subscriptions, right? 😅

Reply:
No, we're not.

But the usage of `loadMultiple` and the fact that `Subscription` is a pretty small and simple entity ought to mean that we can in fact deal with quite the number anyway.

It could be optimized by passing the subscription id instead, and injecting the subscription storage into the queue worker. But I do think this pales in comparison to the CPU cycles we're burning on synchronizing the nodes.